pkg/operation/botanist/operatingsystemconfig.go

```diff
@@ -126,16 +127,25 @@ func (b *Botanist) generateDownloaderConfig(machineImageName string) map[string]
 }
 
 func (b *Botanist) generateOriginalConfig() (map[string]interface{}, error) {
+	var (
+		kubernetesConfig = map[string]interface{}{
+			"clusterDNS": b.Shoot.Networks.CoreDNS.String(),
+			"domain":     gardencorev1beta1.DefaultDomain,
+			"version":    b.Shoot.Info.Spec.Kubernetes.Version,
+		}
+	)
+
+	if b.Shoot.NodeLocalDNSEnabled && b.Shoot.IPVSEnabled() {
+		kubernetesConfig["clusterDNS"] = net.ParseIP("169.254.20.10").String()
```

Comment:
Can you please add a comment above or link [the relevant documentation](https://kubernetes.io/docs/tasks/administer-cluster/nodelocaldns/). Otherwise I find it hard to understand. 

Can we somehow make clear  in the comment that
- IPtables -> node-local-dns interface binds kube-dns service cluster ip & listens/binds  <node-local-address>
- IPVS -> node-local-dns pods listens/binds only on <node-local-address>.  + [the reason you already mentioned](https://github.com/gardener/gardener/pull/2528/files#r447438938) above. Then also state that --cluster-dns flag to kubelet needs to be modified to use <node-local-address> where the NodeLocal DNSCache is listening on.

Reply:
> Does that set the the kubelet flag --cluster-dns pointing to the that NodeLocal DNSCache pod is listening on?
yes. 



> Can you please add a comment above or link [the relevant documentation](https://kubernetes.io/docs/tasks/administer-cluster/nodelocaldns/). Otherwise I find it hard to understand.
> 
> Can we somehow make clear in the comment that
> 
> * IPtables -> node-local-dns interface binds kube-dns service cluster ip & listens/binds
> * IPVS -> node-local-dns pods listens/binds only on .  + [the reason you already mentioned](https://github.com/gardener/gardener/pull/2528/files#r447438938) above. Then also state that --cluster-dns flag to kubelet needs to be modified to use  where the NodeLocal DNSCache is listening on.

Okay will add the link to the usage documenation which should cover your the IPTables and IPVS ... Would that be enough from your perspective?

Reply:
https://github.com/gardener/gardener/pull/2528/files#diff-9c519add19895c6c2c8598c45aa2cf78R32